tests/lib/tools/snapd-state

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: check-state <jq-filter> <comparison> <expected-res> <error-message>"
+    echo "       print-state <jq-filter>"
+    echo "       change-snap-channel <snap-name> <channel>"
+    echo "       force-autorefresh"
+    echo "       prevent-autorefresh"
+    echo "       wait-for-autorefresh <last-change-id>"
+    echo ""
+    echo "The tool is used to manage the snapd state by editing the state.json file."
+    echo "Also it is used to check state and refresh status"
+}
+
+print_state() {
+    JQ_FILTER=$1
+    if [ -z "$JQ_FILTER" ]; then
+        echo "snapd-state: jq filter is needed"
+        exit 1
+    fi
+    # we are storing jq filter specs in JQFILTER, so we can't use --arg like 
+    # normal with jq, so instead just do a _real_ simple and potentially 
+    # _really_ annoying string concatenation with the actual verbatim variable
+    jq -r "$JQ_FILTER" < /var/lib/snapd/state.json
+}
+
+check_state() {
+    JQ_FILTER=$1
+    COMP=$2
+    RES=$3
+    FAIL_MSG="${4:-failed values comparison}"
+    if [ "$COMP" = "!=" ]; then
+        if [ "$(print_state "$JQ_FILTER")" = "$RES" ]; then
+            echo "snapd-state: $FAIL_MSG"
+            exit 1
+        fi
+    elif [ "$COMP" = "=" ]; then
+        if [ "$(print_state "$JQ_FILTER")" != "$RES" ]; then
+            echo "snapd-state: $FAIL_MSG"
+            exit 1
+        fi
+    else
+        echo "snapd-state: invalid comparison operator: $COMP"
+        exit 1
+    fi
+}
+
+change_snap_channel() {
+    local SNAP="$1"
+    local CHANNEL="$2"
+    jq ".data.snaps[\"$SNAP\"].channel = \"$CHANNEL\"" /var/lib/snapd/state.json > /var/lib/snapd/state.json.new
+    mv /var/lib/snapd/state.json.new /var/lib/snapd/state.json
+}
+
+force_autorefresh() {
+    jq ".data[\"last-refresh\"] = \"2007-08-22T09:30:44.449455783+01:00\"" /var/lib/snapd/state.json > /var/lib/snapd/state.json.new
+    mv /var/lib/snapd/state.json.new /var/lib/snapd/state.json
+}
+
+prevent_autorefresh() {
+    jq ".data[\"last-refresh\"] = \"$(date +%Y-%m-%dT%H:%M:%S%:z)\"" /var/lib/snapd/state.json > /var/lib/snapd/state.json.new
+    mv /var/lib/snapd/state.json.new /var/lib/snapd/state.json
+}
+
+wait_for_autorefresh() {
```

Comment:
In #10633 I use slightly modified version of it that waits for any auto-refresh , maybe we should support such variant here as well (e.g. by making $SNAP optional)?

Reply:
Yes I saw that, I think best is to have 2 different commands because the message to check is different, I'll update the current one to make specific for snaps autorefresh.
Then when #10633 is merged, I'll add a new command.
Does it make sense?

Reply:
Yes, sounds good!